src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -841,167 +745,121 @@ public bool EndsWith(char value)
         // Determines whether two strings match.
         public override bool Equals(Object obj)
         {
-            if (object.ReferenceEquals(this, obj))
+            if (ReferenceEquals(this, obj))
+            {
                 return true;
+            }
 
             string str = obj as string;
-            if (str == null)
-                return false;
-
-            if (this.Length != str.Length)
+            if (str is null || this.Length != str.Length)
+            {
                 return false;
+            }
 
             return EqualsHelper(this, str);
         }
 
         // Determines whether two strings match.
-        public bool Equals(String value)
+        public bool Equals(string value)
         {
-            if (object.ReferenceEquals(this, value))
+            if (ReferenceEquals(this, value))
+            {
                 return true;
+            }
 
-            // NOTE: No need to worry about casting to object here.
-            // If either side of an == comparison between strings
-            // is null, Roslyn generates a simple ceq instruction
-            // instead of calling string.op_Equality.
-            if (value == null)
-                return false;
-
-            if (this.Length != value.Length)
+            if (value is null || this.Length != value.Length)
```

Comment:
Why is `value is null` better than `value == null`?

The disassembly is identical.

Reply:
Doesn't affect string as its special cased; but in other cases can cause the compiler to do an implicit conversion to the type then use an overloaded `==` to compare; has caught me out before.